pkg/identity/chainguard/principal.go

```diff
@@ -50,9 +51,14 @@ func PrincipalFromIDToken(_ context.Context, token *oidc.IDToken) (identity.Prin
 		return nil, err
 	}
 
+	subject, err := oauthflow.SubjectFromToken(token)
+	if err != nil {
+		return nil, err
+	}
+
 	return &workflowPrincipal{
 		issuer:           token.Issuer,
-		subject:          token.Subject,
+		subject:          subject,
```

Comment:
We want our subject to be our subject.  It's the challenge that's wrong.

Reply:
I don't know what else keys off of the `principal.Name` but I could see us changing that.

I think that Fulcio and `cosign` using different functions to extract the information being signed for PoP is incredibly fragile because in cases like this the different pieces of logic will disagree even though nothing is actually incorrect (they are just extracting things inconsistently).

Reply:
It looks like this is challenging to call directly the way the principal interface is set up, and the only call site of `Name()` is for the PoP so I can stage a change to do this with some test cases.

Reply:
I would create another field, `popSubject`, set equal to `oauthflow.SubjectFromToken`, and update `Name()` to return `token.popSubject`.

Reply:
@haydentherapper I just saw your comment, that's what my PR does essentially.

Reply:
PR LGTM